### PR TITLE
Allow a Chatty host to be created with source as well as a url

### PR DESCRIPTION
**Pull request by bryans99**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -49,11 +49,13 @@ export class Chatty {
    * for details.
    *
    * @param url The URL of the client iframe to create. The hosted iframe should create a chatty
-   * client to communicate with the host.
+   * client to communicate with the host. Optional, if not provided, source will be used,
+   * @param source The source of the client iframe to create. The hosted iframe should create a chatty
+   * client to communicate with the host. Otional, if not provided, url will be used.
```
Comment (whscullin) [inline src/index.ts]: p

File: src/index.ts [hunk 2]
```diff
@@ -49,11 +49,13 @@ export class Chatty {
...
    */
 
-  static createHost (url: string) {
-    return new ChattyHostBuilder(url)
+  static createHost (url?: string, source?: string) {
+    return new ChattyHostBuilder(url, source)
```
Comment (whscullin) [inline src/index.ts]: I might prefer a second `createHostFromSource()` method than one with an "optional" null parameter.

File: src/index.ts (hunk 1 above)
Comment (bryans99) [reply src/index.ts]: p?

File: src/index.ts (hunk 2 above)
Comment (bryans99) [reply src/index.ts]: Dammit, thats what I did originally :D 

I will change it.

Thanks

**Pull request merged**